test/simulator/moves/thousandarrows.js

```diff
@@ -29,31 +29,31 @@ describe('Thousand Arrows', function () {
 		assert.strictEqual(battle.p2.active[0].boosts.spa, 2);
 	});
 
-	it.skip('should not ignore type effectiveness on the first hit against Flying-type Pokemon with Ring Target', function () {
+	it('should not ignore type effectiveness on the first hit against Flying-type Pokemon with Ring Target', function () {
 		battle = BattleEngine.Battle.construct();
 		battle.join('p1', 'Guest 1', 1, [{species: "Zygarde", level: 10, ability: 'aurabreak', item: 'laggingtail', moves: ['thousandarrows']}]);
 		battle.join('p2', 'Guest 2', 1, [{species: "Ho-Oh", ability: 'wonderguard', item: 'ringtarget', moves: ['recover']}]);
 		battle.commitDecisions();
-		assert.notStrictEqual(battle.p2.active.hp, battle.p2.active.maxhp);
+		assert.notStrictEqual(battle.p2.active[0].hp, battle.p2.active[0].maxhp);
 	});
 
-	it.skip('should have its effects against Flying-type Pokemon rendered null by Iron Ball', function () {
+	it('should not ground or deal neutral damage to Flying-type Pokemon holding an Iron Ball', function () {
 		battle = BattleEngine.Battle.construct();
 		battle.join('p1', 'Guest 1', 1, [{species: "Zygarde", level: 10, ability: 'aurabreak', item: 'laggingtail', moves: ['thousandarrows', 'mudslap']}]);
-		battle.join('p2', 'Guest 2', 1, [{species: "Ho-Oh", ability: 'wonderguard', item: 'ironball', moves: ['recover', 'trick']}]);
+		battle.join('p2', 'Guest 2', 1, [{species: "Ho-Oh", ability: 'shellarmor', item: 'ironball', moves: ['recover', 'trick']}]);
 		battle.commitDecisions();
 		assert.ok(!battle.log[battle.lastMoveLine + 1].startsWith('|-supereffective|'));
 		var hp = battle.p2.active[0].hp;
 		assert.notStrictEqual(hp, battle.p2.active[0].maxhp);
-		battle.choose('p1', 'move 1');
+		battle.choose('p1', 'move 2');
 		battle.choose('p2', 'move 2');
 		assert.strictEqual(hp, battle.p2.active[0].hp);
 	});
```

Comment:
We're checking to make sure Thousand Arrows didn't actually ground the Pokemon that was holding Iron Ball previously. How are we supposed to check that it's regained its Ground immunity if we pick the move that's specifically designed to hit Pokemon that are immune to Ground?